ledger/participant-integration-api/src/main/resources/db/migration/postgres/V50__Append_only_schema.sql

```diff
@@ -49,124 +49,415 @@ CREATE TABLE parameters (
+
+-- lookup divulgance events, in order of ingestion
+CREATE INDEX participant_events_divulgence_contract_id_idx ON participant_events_divulgence USING btree (contract_id, event_sequential_id);
+
+
+
+
+---------------------------------------------------------------------------------------------------
+-- Events table: create
+---------------------------------------------------------------------------------------------------
+CREATE TABLE participant_events_create (
     -- * event identification
-    event_sequential_id bigserial NOT NULL, -- TODO temporarily readding bigserial for original write paths
+    event_sequential_id bigserial NOT NULL, -- TODO append-only: temporarily readding bigserial for original write paths
     -- NOTE: this must be assigned sequentially by the indexer such that
     -- for all events ev1, ev2 it holds that '(ev1.offset < ev2.offset) <=> (ev1.event_sequential_id < ev2.event_sequential_id)
 
-    event_offset bytea,                                 -- NULL for divulgence events
+    event_offset bytea NOT NULL,
 
     -- * transaction metadata
-    transaction_id text,                                -- NULL for migrated divulgence events
-    ledger_effective_time timestamp without time zone,  -- NULL for migrated divulgence events
+    transaction_id text NOT NULL,
+    ledger_effective_time timestamp without time zone NOT NULL,
     command_id text,
     workflow_id text,
     application_id text,
     submitters text[],
 
     -- * event metadata
-    node_index integer,  -- NULL for migrated divulgence events
-    event_id text,       -- NULL for migrated divulgence events
-    -- string representation of (transaction_id, node_index)
+    node_index integer NOT NULL,
+    event_id text NOT NULL,        -- string representation of (transaction_id, node_index)
 
     -- * shared event information
     contract_id text NOT NULL,
-    template_id text,     -- TODO @simon@ with the new divulgance model supporting public pkv implementations: we need this to enable NULL-s. Do we need to make involved indexes partial?
+    template_id text NOT NULL,
     flat_event_witnesses text[] DEFAULT '{}'::text[] NOT NULL,       -- stakeholders of create events and consuming exercise events
     tree_event_witnesses text[] DEFAULT '{}'::text[] NOT NULL,       -- informees for create, exercise, and divulgance events
 
     -- * divulgence and create events
-    create_argument bytea,
+    create_argument bytea NOT NULL,
 
     -- * create events only
-    create_signatories text[],
-    create_observers text[],
+    create_signatories text[] NOT NULL,
+    create_observers text[] NOT NULL,
     create_agreement_text text,
     create_key_value bytea,
     create_key_hash bytea,
 
+    -- * compression flags
+    create_argument_compression SMALLINT,
+    create_key_value_compression SMALLINT
+);
+-- these columns contain data that is generally incompressible, so don't try it
```

Comment:
Can you update this comment? It's quite ambiguous

Reply:
Updated the comment.